### PR TITLE
Rewired FSM to wait until Mech Deliver received

### PR DESCRIPTION
**Pull request by jmoreira-valory**
Rewired FSM to wait until Mech Deliver received

### CODE REVIEW

**Review by angrybayblade — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

File: packages/valory/skills/decision_maker_abci/rounds.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ class DecisionMakerAbciApp(AbciApp[Event]):
         },
         DecisionReceiveRound: {
             Event.DONE: BetPlacementRound,
-            Event.MECH_RESPONSE_ERROR: BlacklistingRound,
+            Event.MECH_RESPONSE_ERROR: DecisionReceiveRound,  # loop on the same state until Mech deliver is received
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/rounds.py:130]: What about the round timeout event?

Comment (jmoreira-valory) [reply packages/valory/skills/decision_maker_abci/rounds.py:130]: To my understanding, round timeout can be caused due to several other issues not only Mech not responding (for example, network latency, slow agent, etc.). What do you suggest to do in this cae?

Comment (DavidMinarsch) [reply packages/valory/skills/decision_maker_abci/rounds.py:130]: Isn't re-entering `DecisionReceiveRound` exactly what we are doing here?

Comment (Adamantios) [reply packages/valory/skills/decision_maker_abci/rounds.py:130]: Yes, what I suggest is that we do the same in case of a round timeout event.

**Pull request merged**